Dialog/TilesetEditDialog.cpp

```diff
@@ -1184,6 +1230,31 @@ void TilesetEditDialog::on_pushButton_ImportPalette_clicked()
             tilesetEditParams->newTileset->SetColor(SelectedPaletteId, j, newcolor.rgba());
         }
     }
+
+    else if (selectedfilter.compare("Raw Binary palette (*.bin)") == 0)
+    {
+         QByteArray tmppalettedata;
+         QFile palbinfile(qFilePath);
+         if(!palbinfile.open(QIODevice::ReadOnly))
+         {
+             QMessageBox::critical(this, QString("Error"), QString("Cannot open file!"));
+             return;
+         }
+         tmppalettedata = palbinfile.readAll();
+         palbinfile.close();
+
+         QVector<QRgb> tmppalette;
+         unsigned short *tmppaldata = new unsigned short[16];
+         memset(tmppaldata, 0, 32);
+         memcpy(tmppaldata, tmppalettedata.data(), qMin(32, tmppalettedata.size()));
+         ROMUtils::LoadPalette(&tmppalette, tmppaldata, true);
+         tmppalette.pop_front(); // Skip the first color
```

Comment:
Instead of pop() and front(), can we please use tmppalettedata[i] in the loop? Having to track the state of this vector would make debugging difficult if there's ever issues in the future

Reply:
:heavy_check_mark: